Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -188,13 +214,27 @@ private class MethodBuilder(val paramNames: MutableCollection<List<String>>): Me
 fun buildJ2ObjcNativeIndex(jarFiles: List<String>): IndexerResult {
   val jarFile = JarFile(jarFiles[0])
 
-  val j2objcClasses = jarFile.use { it.entries().iterator().asSequence()
-    .filter{ it.name.endsWith(".class")}.map {
-      val parser = J2ObjCParser()
-      ClassReader(jarFile.getInputStream(it).readBytes()).accept(parser,0)
-      parser.buildClass()
-    }.toList()
+  val j2objcClasses = mutableListOf<ObjCClass>()
+  var j2objcProtocols = mutableListOf<ObjCProtocol>()
+
+  val protocolClasses = jarFile.entries().iterator().asSequence()
+    .filter{ it.name.endsWith(".class")}
+  for (classFile in protocolClasses)
+  {
+    val parser = J2ObjCParser()
+    ClassReader(jarFile.getInputStream(classFile).readBytes()).accept(parser,0)
+    if (parser.access and Opcodes.ACC_INTERFACE != 0)
+      j2objcProtocols.add(parser.buildInterface())
+  }
+  val classClasses = jarFile.entries().iterator().asSequence()
+    .filter{ it.name.endsWith(".class")}
+  for (classFile in classClasses)
+  {
+    val parser = J2ObjCParser()
+    ClassReader(jarFile.getInputStream(classFile).readBytes()).accept(parser,0)
+    if (parser.access and Opcodes.ACC_INTERFACE == 0)
+      j2objcClasses.add(parser.buildClass(j2objcProtocols))
```

Comment:
I need to parse through the class files twice, once in order to find interfaces and define them, the second time to find classes and attach the interface definitions to them, couldn't figure out a better way.

Question: How should I actually design this, sequences can only be used once apparently, so I'm trying to figure out a good way of doing this.